WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -774,6 +781,25 @@ private extension ProductFormViewController {
         SharingHelper.shareURL(url: url, title: product.name, from: view, in: self)
     }
 
+    func duplicateProduct() {
+        showSavingProgress(.duplicate)
+        viewModel.duplicateProduct(onCompletion: { [weak self] result in
+            switch result {
+            case .failure(let error):
+                DDLogError("⛔️ Error duplicating Product: \(error)")
+
+                // Dismisses the in-progress UI then presents the error alert.
+                self?.navigationController?.dismiss(animated: true) {
+                    self?.displayError(error: error)
+                }
+            case .success:
+                // Dismisses the in-progress UI, then presents the confirmation alert.
+                self?.navigationController?.dismiss(animated: true, completion: nil)
+                self?.presentProductConfirmationSaveAlert(hasDuplicated: true)
```

Comment:
❓  Should we present the alert in the completion block as the comment explains? 
```suggestion
                self?.navigationController?.dismiss(animated: true) {
                    self?.presentProductConfirmationSaveAlert(hasDuplicated: true)
                }
```